pkg/util/priorityqueue/priorityqueue.go

```diff
@@ -0,0 +1,92 @@
+// Copyright 2021 VMware, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package priorityqueue
+
+import (
+	"container/heap"
+	"time"
+
+	"github.com/vmware/go-ipfix/pkg/entities"
+	"github.com/vmware/go-ipfix/pkg/util"
+)
+
+type ItemToExpire struct {
+	// Flow related info
+	FlowKey             util.FlowKey
+	FlowRecord          entities.Record
+	IsValid             bool
```

Comment:
I'm curious as to why we don't just keep a pointer to the aggregationRecord here?
it seems that because we have a single mutex it would work fine.

Reply:
I wanted to reuse this code for Antrea flow exporter, so moved it to the `entitied.Record` interface. Considering the fact that we maintain a separate record data structure in Antrea and a reference to that has to be stored in the priority queue, maybe keeping this code separate for aggregation process is better.

Reply:
Stored the pointer to the Aggregation record directly and kept the flow key as well, because it is slightly costly to get the flow key from the aggregation record again. Keeping this code specific to the aggregation process for now.